osu.Game/Screens/Edit/Editor.cs

```diff
@@ -966,6 +966,7 @@ private void updateLastSavedHash()
         {
             new EditorMenuItem(WebCommonStrings.ButtonsSave, MenuItemType.Standard, () => Save()),
             new EditorMenuItem(EditorStrings.ExportPackage, MenuItemType.Standard, exportBeatmap) { Action = { Disabled = !RuntimeInfo.IsDesktop } },
+            new EditorMenuItem(EditorStrings.ExportLegacyPackage, MenuItemType.Standard, exportLegacyBeatmap) { Action = { Disabled = !RuntimeInfo.IsDesktop } },
```

Comment:
Again, unsure on this being a separate archive type and all. I think outputting one archive, but with like an `.osu2` inside for lazer could work better for users, because this right now just kinda feels really ugly.

I guess the issue with that other solution is that if someone exports from lazer, then edits in stable and reimports that after the stable edits, lazer may see the `.osu2` file which stable didn't touch, therefore ignoring the changes from stable. Could compensate for it by adding _another_ file which lazer would write, basically something like:

```json
{"osu_hash":"${SHA2_OF_OSU_FILE}","osu2_hash":"${SHA2_OF_OSU2_FILE"}
```

which helps, because if lazer sees that, and then sees an `.osu` file that doesn't match the hash, then it can figure out which thing should take precedence.

Dunno, interested in more opinions. The main point I'm driving at is that I'd like to hide as much of this ugliness from users as possible.

Reply:
This is what I wanted to do, but I believe @OliBomby had reasons for not wanting to implement things this way. Will wait to hear his view on it these days.

Reply:
My opinion on this hasn't changed. I think it's better to have an explicit export option for legacy osu! because it will free the lazer file format from the constraints if osu stable, allowing for more changes to be made to the lazer file format without breaking compatibility with stable.

Considering that I predict the lazer file format will change more in the future, it doesnt make sense to use some half-legacy half-lazer Frankenstein's file format. Therefore I think the explicit is the simpler and more elegant solution.

Even if the lazer file format doesn't change, I think this explicit export will be easier to understand for users because it is just a simpler approach. The exported packages wont have any weird doubled .osu2 files and hashes which prompt the mapper to do a 10 hour investigation on what is really going on.

Reply:
Okay, well, I don't feel _that_ strongly about a separate export option, but if we do end up having it, can we not have it as two top-level menu items and instead do something like

![image](https://github.com/ppy/osu/assets/20418176/2e82d196-6ad6-4975-aa7a-728bba63195b)

(copy up for discussion, obviously)?

Reply:
I like that. I'll add the submenu